From 47275ee115a0503a5e9747bda50c580b06537015 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 19 Oct 2016 20:32:34 +0100
Subject: [PATCH 4/6] panel: Use an actual timestamp when focusing the window,
 not 0

GDK_CURRENT_TIME is actually set to 0, and only gtk_window_present interprets
it as something useful, by passing the X11 display current time. This causes
numerous issues in budgie-wm, so don't do it.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 panel/panel.vala | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/panel/panel.vala b/panel/panel.vala
index 812daff..be4dcb7 100644
--- a/panel/panel.vala
+++ b/panel/panel.vala
@@ -986,8 +986,14 @@ public class Panel : Budgie.Toplevel
 
         if (expanded) {
             Idle.add(()=> {
-                if (get_window() != null) {
-                    get_window().focus(Gdk.CURRENT_TIME);
+                Gdk.Window? panel_window = get_window();
+                if (panel_window != null) {
+                    Gdk.Display? display = screen.get_display();
+                    if (display is Gdk.X11Display) {
+                        panel_window.focus(Gdk.X11Display.get_user_time(display));
+                    } else {
+                        panel_window.focus(Gtk.get_current_event_time());
+                    }
                 }
                 present();
                 return false;
-- 
2.10.1

